src/CurrencyControl/index.tsx

```diff
@@ -13,6 +13,11 @@ interface CurrencyControlProps extends CurrencyInputProps {
     help?: string;
 }
 
+const formatter = new Intl.NumberFormat(navigator.language);
+const groupSeparator = formatter.format(1000).replace(/[0-9]/g, '');
+const decimalSeparator = formatter.format(1.1).replace(/[0-9]/g, '');
```

Comment:
Hey @JoshuaHungDinh, I’m just commenting on this because I worked on something similar, and there’s an approach that uses less “magic” than RegEx:

```ts
const numberFormat = new Intl.NumberFormat(window.navigator.language);
const parts = numberFormat.formatToParts(1234.56);

const groupSeparator = parts.find(part => part.type === 'group')?.value ?? '';
const decimalSeparator = parts.find(part => part.type === 'decimal')?.value ?? '';
```

Reply:
Hey @pauloiankoski thanks!

How's this looking? 46c32a85bc2fa5411c03fe0e21d3c49e589ed3c3